ccmlib/utils/sni_proxy.py

```diff
@@ -0,0 +1,169 @@
+        config_file.write(yaml.safe_dump(config, sort_keys=False))
+
+    config = dict(datacenters={'eu-west-1': dict(certificateAuthorityPath=os.path.join(ssl_dir, 'ccm_node.cer'),
+                                                 server=f'{host}:{port}',
+                                                 nodeDomain='cluster-id.scylla.com')},
+                  authInfos={'default': dict(clientCertificatePath=os.path.join(ssl_dir, 'ccm_node.cer'),
+                                             clientKeyPath=os.path.join(ssl_dir, 'ccm_node.key'),
+                                             username=username,
+                                             password=password,
+                                             insecureSkipTlsVerify=False)},
+                  contexts={'default': dict(datacenterName='eu-west-1', authInfoName='default')},
+                  currentContext='default')
+
+    with open(os.path.join(ssl_dir, 'config_path.yaml'), 'w') as config_file:
+        config_file.write(yaml.safe_dump(config, sort_keys=False))
+
+    return os.path.join(ssl_dir, 'config_data.yaml'), os.path.join(ssl_dir, 'config_path.yaml')
+
+
+def stop_sni_proxy(docker_id):
+    subprocess.check_output(['/bin/bash', '-c', f'docker rm -f {docker_id}'])
+
+
+def configure_sni_proxy(conf_dir, nodes_info, listen_port=443):
+    sniproxy_conf_tmpl = dedent("""
+        user sniproxy
+        pidfile /var/run/sniproxy/sniproxy.pid
+
+        error_log {
+          filename /dev/stderr
+          priority debug
+        }
+
+        listener $FIRST_ADDRESS $listen_port {
+          proto tls
+          access_log {
+            filename /dev/stdout
+          }
+        }
+
+        table {
+        $TABLES
+        }
+        """)
+    tables = ""
+    mapping = {}
+    address, port, host_id = list(nodes_info)[0]
+    tables += f"  any.cluster-id.scylla.com {address}:{port}\n"
+    mapping['FIRST_ADDRESS'] = address
+    mapping['listen_port'] = listen_port
+
+    for address, port, host_id in nodes_info:
+        tables += f"  {host_id}.cluster-id.scylla.com {address}:{port}\n"
+
+    tmpl = string.Template(sniproxy_conf_tmpl)
+    sniproxy_conf_path = os.path.join(conf_dir, 'sniproxy.conf')
+
+    with open(sniproxy_conf_path, 'w') as fp:
+        fp.write(tmpl.substitute(TABLES=tables, **mapping))
+
+    return sniproxy_conf_path
+
+
+def start_sni_proxy(conf_dir, nodes_info, listen_port=443):
+    address, _, _ = list(nodes_info)[0]
+    sniproxy_conf_path = configure_sni_proxy(conf_dir, nodes_info, listen_port=443)
+    sniproxy_dockerfile = os.path.join(os.path.dirname(__file__), '..', 'resources', 'docker', 'sniproxy')
+    subprocess.check_output(['/bin/bash', '-c', f'docker build {sniproxy_dockerfile} -t sniproxy'], universal_newlines=True)
+    docker_id = subprocess.check_output(['/bin/bash', '-c', f'docker run -d --network=host -v {sniproxy_conf_path}:/etc/sniproxy.conf -p 443 -it sniproxy'], universal_newlines=True)
```

Comment:
```suggestion
    docker_id = subprocess.check_output(['/bin/bash', '-c', f'docker run -d --network=host -v {sniproxy_conf_path}:/etc/sniproxy.conf:z -p 443 -it sniproxy'], universal_newlines=True)
```

Without that `:z` the container fails to start on my computer, evidenced by `sniproxy` logs:
```
2022-06-20 16:15:59 init_config: unable to open configuration file: /etc/sniproxy.conf
Unable to load /etc/sniproxy.conf
```
The open fails with ERR=13 (Permission denied). (quickly debugged with [opensnoop](https://github.com/iovisor/bcc/blob/master/tools/opensnoop.py))

Reply:
fixed